gadget-container/entrypoint.sh

```diff
@@ -98,7 +98,10 @@ if [ "$ID" = "rhcos" ] && [ ! -d "/usr/src/kernels/$KERNEL" ]; then
     RPM=kernel-devel-$KERNEL.rpm
 
     mkdir -p $RPMDIR/usr/src/kernels/
-    curl -fsSLo $RPMDIR/$RPM $REPO/$RPM || true
+    # Ensure the whole operation does not take more than 40 seconds. This value
+    # needs to be lower than the liveness-probe initial delay. Otherwise, there
+    # is the risk the gadget pod gets restarted while downloading this package.
+    curl --max-time 40 -fsSLo $RPMDIR/$RPM $REPO/$RPM || true
```

Comment:
When curl exits because of the max-time, the file is partially downloaded. Since you use `|| true`, it wiwll continue to execute. Wouldn't rpm2cpio fail below? Is it the expected behaviour?

Reply:
That's true. I think I should remove the file if exists with an error so that the script does not try to uncompress it. WDYT?

Reply:
Sounds good!

Reply:
I updated the PR. 

For testing, I forced it to fail and I got the following logs and the pod continue running without problems:

```bash
Fetching kernel-devel from CentOS Vault Mirror...
curl: (28) Operation timed out after 1000 milliseconds with 10398415 out of 1073741824 bytes received
Failed to fetch kernel-devel package http://vault.centos.org/8.4.2105/BaseOS/x86_64/os/Packages/kernel-devel-4.18.0-305.19.1.el8_4.x86_64.rpm
```